app/models/product_importer.rb

```diff
@@ -177,7 +181,12 @@ def count_existing_products
           AND spree_variants.deleted_at IS NULL', supplier_id).
           count
 
-        @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        if @supplier_products[:by_supplier][supplier_id]
+          @supplier_products[:by_supplier][supplier_id][:existing_products] = products_count
+        else
+          @supplier_products[:by_supplier][supplier_id] = {existing_products: products_count}
+        end
```

Comment:
I've refactored it a bit to make it slightly clearer. I split part of the data out into a separate attribute, simplified the hash and renamed the keys.